src/extension.ts

```diff
@@ -97,6 +97,7 @@ let allHotspotsView: VSCode.TreeView<HotspotTreeViewItem>;
 let helpAndFeedbackTreeDataProvider: HelpAndFeedbackTreeDataProvider;
 let helpAndFeedbackView: VSCode.TreeView<HelpAndFeedbackLink>;
 let taintVulnerabilityCollection: VSCode.DiagnosticCollection;
+let currentProgress: { progress: VSCode.Progress<{ increment?: number }>, resolve: () => void } | undefined;
```

Comment:
Having a single progress at a time may be dangerous. How about having a map indexed on the `taskId` from the LSP methods?

Reply:
Yes I was planning to tackle this. Did not go forward since I was not sure if LS part was OK. I will fix it.